### PR TITLE
Adds support for schema variables

### PR DESCRIPTION
**Pull request by turekj**
🏭 Adds support for #236. Allows to define variables per schema configuration or per target.

To a future reviewer: thanks in advance, you are awesome! 👍 🍻

### CODE REVIEW

File: Docs/ProjectSpec.md [hunk 1]
```diff
@@ -447,6 +451,11 @@ A multiline script can be written using the various YAML multiline methods, for
 - [ ] **gatherCoverageData**: **Bool** - a boolean that indicates if this scheme should gather coverage data. This defaults to false
 - [ ] **targets**: **[String]** - a list of targets to test
 
+### Environment Variable
+- [x] **key**: **String** - variable's key.
```
Comment (yonaskolb) [inline Docs/ProjectSpec.md]: Maybe `variable` is a better name here. The yaml key can be changed here, but maybe also the property name in https://github.com/xcodeswift/xcproj/pull/227?
What do you think?

Comment (turekj) [reply Docs/ProjectSpec.md]: I have updated that both in xcproj and XcodeGen! 💯

**Comment by turekj** [comment]
Hi @yonaskolb, I did all the requested updates. 

Unfortunately, I could not make integration tests pass on my local machine after updating `xcproj` version to a master one. The tests fail with `EXC_BAD_ACCESS` with a too deep recursion nest level. This behaviour is reproducible on a master branch with `xcproj` dependency bumped to master branch of the main repository.

**Comment by yonaskolb** [comment]
@turekj thanks for highlighting the recursion issue in xcproj. I've fixed it here https://github.com/xcodeswift/xcproj/pull/237
One the next xcproj release is done we can merge this 👍

**Comment by yonaskolb** [comment]
xxproj 4.1.0 is now released

**Comment by yonaskolb** [comment]
@turekj, are you able to update the xcproj package, so we can get this merged? If not, or I don't hear back by tomorrow I'll merge and then fix in master 👍

**Comment by turekj** [comment]
@yonaskolb I will handle that today/tomorrow! 👍

**Comment by turekj** [comment]
@yonaskolb ready to go! 🎉

**Pull request merged**